### PR TITLE
tx_list fix + tx_info collateral and performance boost

### PR DESCRIPTION
**Pull request by Scitz0**
- Fix for tx_list that didn't filter out pools correctly based on retire status.
- Collateral inputs added to tx_info endpoint + koiosapi spec update
- tx_info rewritten to not join on CTE but instead select on it in a subquery. This gives a big performance boost as CTEs are inefficient on joins. Quick test showed ~6-8x performance boost.

### CODE REVIEW

File: files/grest/rpc/pool/pool_list.sql [hunk 1]
```diff
@@ -8,19 +8,32 @@ CREATE FUNCTION grest.pool_list ()
...
-WHERE
-  pic.pool_status != 'retired'
-ORDER BY
-  pic.pool_id_bech32,
-  pic.tx_id DESC;
-END;
-$$;
 
-COMMENT ON FUNCTION grest.pool_list IS 'A list of all currently registered/retiring (not retired) pools';
+  RETURN QUERY (
```
Comment (rdlrt) [inline files/grest/rpc/pool/pool_list.sql:12]: can we change this to return table format? (consistency as well as unlocks full potential for PostgREST API)

PS: There is a seperate ticket for this request, so can be looked at seperately for almost all pool queries

Comment (DamjanOstrelic) [reply files/grest/rpc/pool/pool_list.sql:12]: I am confused - the function is defined as `RETURNS TABLE`. I thought we have to `RETURN QUERY` in the actual function and `RETURN QUERY` actually returns a table?

Comment (rdlrt) [reply files/grest/rpc/pool/pool_list.sql:12]: ah - my bad, I misread it

**Review by DamjanOstrelic — APPROVED** [review]
Tested on guild network 👍

**Review by rdlrt — APPROVED** [review]

**Pull request merged**